---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# GLP: Feature Selection for scRNA-seq Analyses

<!-- badges: start -->
[![R-CMD-check](https://github.com/gddhouse/GLP/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/gddhouse/GLP/actions/workflows/R-CMD-check.yaml)
[![Codecov test coverage](https://codecov.io/gh/gddhouse/GLP/graph/badge.svg)](https://app.codecov.io/gh/gddhouse/GLP)
<!-- badges: end -->

GLP is an R package designed for efficient feature selection in single-cell RNA sequencing (scRNA-seq) analyses. It identifies highly variable genes (HVGs) by analyzing the relationship between the positive ratio and average expression level. The package uses optimized LOESS regression with automatic bandwidth selection based on the Bayesian Information Criterion (BIC) and applies Tukeyâ€™s biweight method to reduce the impact of outliers. GLP offers improved accuracy in gene selection, helping to preserve essential biological information for downstream analyses.

## Installation

You can install the development version of GLP from [GitHub](https://github.com/) with:

``` r
# install.packages("pak")
pak::pak("gddhouse/GLP")
```

## Example

This is a basic example which shows you how to solve a common problem:

```{r example}
library(GLP)
## basic example code
```

What is special about using `README.Rmd` instead of just `README.md`? You can include R chunks like so:

```{r cars}
summary(cars)
```

You'll still need to render `README.Rmd` regularly, to keep `README.md` up-to-date. `devtools::build_readme()` is handy for this.

You can also embed plots, for example:

```{r pressure, echo = FALSE}
plot(pressure)
```

In that case, don't forget to commit and push the resulting figure files, so they display on GitHub and CRAN.
